vermouth/molecule.py

```diff
@@ -600,16 +615,31 @@ def to_molecule(self, atom_offset=0, resid=1, offset_charge_group=1):
 class Link(Block):
     """
     Template link between two residues.
+
+    Parameters
+    ----------
+    incoming_graph_data:
+        Data to initialize graph. If None (default) an empty graph is created.
+    attr:
+        Attributes to add to graph as key=value pairs.
     """
     node_dict_factory = OrderedDict
 
-    def __init__(self):
-        super().__init__()
-        self.non_edges = []
-        self.removed_interactions = {}
+    def __init__(self, incoming_graph_data=None, **attr):
+        super().__init__(incoming_graph_data, **attr)
+        # Arbitrary attributes can be set during the initialization. We need
+        # to set the default of some key attributes, but without overwritting
+        # what has been passed in the 'attr' argument.
+        defaults = {
+            'non_edges': [],
+            'removed_interactions': {},
+            'molecule_meta': {},
+            'patterns': [],
+        }
+        for attribute, default_value in defaults.items():
+            if not hasattr(self, attribute):
+                setattr(self, attribute, default_value)
```

Comment:
We can wrap the whole default values things in a class, and do some magical inheritance if we want to cut back on code duplication. Or make it a function, if you're not feeling adventurous.